src/libstd/sync/mpsc/spsc_queue.rs

```diff
@@ -84,18 +95,60 @@ impl<T> Queue<T> {
     ///               no bound. Otherwise, the cache will never grow larger than
     ///               `bound` (although the queue itself could be much larger.
     pub unsafe fn new(bound: usize) -> Queue<T> {
+        Self::with_additions(bound, (), ())
```

Comment:
@bors r-

This method is never used on asm.js (Emscripten), causing an unused warning when testing `stage2-libstd`.

```rust
[02:11:04] Testing libstd stage2 (x86_64-unknown-linux-gnu -> asmjs-unknown-emscripten)
[02:11:04]    Compiling rand v0.0.0 (file:///checkout/src/librand)
[02:11:04]    Compiling std_unicode v0.0.0 (file:///checkout/src/libstd_unicode)
[02:11:04]    Compiling alloc v0.0.0 (file:///checkout/src/liballoc)
[02:11:04]    Compiling core v0.0.0 (file:///checkout/src/libcore)
[02:11:12]    Compiling collections v0.0.0 (file:///checkout/src/libcollections)
[02:11:21]    Compiling std v0.0.0 (file:///checkout/src/libstd)
[02:11:50] error: method is never used: `new`
[02:11:50]   --> /checkout/src/libstd/sync/mpsc/spsc_queue.rs:97:5
[02:11:50]    |
[02:11:50] 97 |     pub unsafe fn new(bound: usize) -> Queue<T> {
[02:11:50]    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[02:11:50]    |
[02:11:50] note: lint level defined here
[02:11:50]   --> /checkout/src/libstd/lib.rs:232:9
[02:11:50]    |
[02:11:50] 232| #![deny(warnings)]
[02:11:50]    |         ^^^^^^^^
[02:11:50]    = note: #[deny(dead_code)] implied by #[deny(warnings)]
[02:11:50] 
[02:11:51] error: aborting due to previous error
[02:11:51] 
[02:11:52] error: Could not compile `std`.
[02:11:52] warning: build failed, waiting for other jobs to finish...
[02:12:49] error: build failed
[02:12:49] 
[02:12:49] 
[02:12:49] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "test" "--target" "asmjs-unknown-emscripten" "-j" "4" "--release" "--locked" "--color" "always" "--features" "panic-unwind jemalloc backtrace" "--manifest-path" "/checkout/src/libstd/Cargo.toml" "-p" "std:0.0.0" "-p" "std_unicode:0.0.0" "-p" "alloc:0.0.0" "-p" "panic_abort:0.0.0" "-p" "rand:0.0.0" "-p" "compiler_builtins:0.0.0" "-p" "unwind:0.0.0" "-p" "core:0.0.0" "-p" "libc:0.0.0" "-p" "collections:0.0.0" "-p" "alloc_system:0.0.0" "--"
[02:12:49] expected success, got: exit code: 101
[02:12:49] 
[02:12:49] 
[02:12:49] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test --target asmjs-unknown-emscripten
[02:12:49] Build completed unsuccessfully in 2:10:23
```

Reply:
I've cfg'd it out for emscripten in 41320fa